src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/RestIndexAnomalyDetectorAction.java

```diff
@@ -152,6 +152,13 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
     public List<Route> routes() {
         return ImmutableList
             .of(
+                // Create (legacy)
```

Comment:
I think we want a new `Route` constructor that takes multiple paths here to make it easier on the plugin authors.

So 

```
new Route(
  RestRequest.Method.PUT,
  ImmutableList.of(
    String.format(Locale.ROOT, "%s/{%s}", AnomalyDetectorPlugin.AD_BASE_LEGACY_DETECTORS_URI, DETECTOR_ID),
    String.format(Locale.ROOT, "%s/{%s}", AnomalyDetectorPlugin.AD_BASE_DETECTORS_URI, DETECTOR_ID)
), 
 ...
```

Reply:
I understand we are duplicating code.
Looking at `Route` its seems like it was built to handle only a single path. It would need some changes across OpenSearch to accept multiple paths and also probably it could be confusing to have two different paths in a single Route. 

May what I could do is have util function which will take a list of base path's and API paths, return a list of routes.
What do you think?

Reply:
Totally agree. A static method on `Route` that returns a list of routes sounds from a matrix of paths and methods sounds right!